core/src/main/java/io/opencensus/stats/StatsManager.java

```diff
@@ -32,5 +32,5 @@
   /**
    * Returns the default {@link StatsContextFactory}.
    */
-  abstract StatsContextFactory getStatsContextFactory();
+  public abstract StatsContextFactory getStatsContextFactory();
```

Comment:
> Do you think that registering/querying views and recording stats should also be separate? I'm not sure how we can separate those parts, since they need to work with shared state. If people inject them separately, the objects would still need to come from the same StatsManagerFactory instance (or some root object with a better name) that can ensure that the objects are using the same state.

I understand in production recording and querying must share state. I think `Stats.java` is taking care of that. For the sake of unit testing, recording and querying do need to be separately injected, because one application component only deal with one of them.

> One idea is to call the singleton StatsComponent and give it methods getViewManager() and getStatsRecorder(). Then you could get the ViewManager and StatsRecorder from the singleton and pass them around separately. Would that address the problem with bundling unrelated functionalities?

That would work. Since you will have `StatsRecorder()`, will you also deprecate the `record()` on `StatsContext`?